### PR TITLE
Fixes #1659 by testing both 8 and 9 for water blockid

### LINKED ISSUE #1659: Ocean floor covers legitimate caves when using the Cave render mode
The ocean floor blocks the view of many legitimate caves when using the Cave render mode. It would be much nicer if the ocean floor was not considered part of a cave, especially if your map is dominated by the different ocean biomes like mine is. You can see in my attached image how it is not exactly great to look at. If there is a way to filter out the ocean floor, then it should be added to the docs. If it is already there then I should learn how to read.
Selection_109

### PR DESCRIPTION
**Pull request by dgrimreaper**

### CODE REVIEW

File: overviewer_core/src/primitives/cave.c [hunk 1]
```diff
@@ -78,15 +78,15 @@ cave_hidden(void* data, RenderState* state, int32_t x, int32_t y, int32_t z) {
      * at this point of the code the block has no skylight
      * but a deep sea can be completely dark
      */
-    if ((getArrayShort3D(state->blocks, x, y, z) == 9) ||
-        (get_data(state, BLOCKS, x, y + 1, z) == 9)) {
 
+    if ((getArrayShort3D(state->blocks, x, y, z) == 9) || (getArrayShort3D(state->blocks, x, y, z) == 8) ||
+        (get_data(state, BLOCKS, x, y + 1, z) == 9) || (get_data(state, BLOCKS, x, y + 1, z) == 8)) {
```
Comment (CounterPillow) [inline overviewer_core/src/primitives/cave.c]: Is the compiler smart enough to optimise away those repeated function calls?

File: overviewer_core/src/primitives/cave.c [hunk 2]
```diff
@@ -78,15 +78,15 @@ cave_hidden(void* data, RenderState* state, int32_t x, int32_t y, int32_t z) {
...
         for (dy = y + 1; dy < (SECTIONS_PER_CHUNK - state->chunky) * 16; dy++) {
             /* go up and check for skylight */
             if (get_data(state, SKYLIGHT, x, dy, z) != 0) {
                 return true;
             }
-            if (get_data(state, BLOCKS, x, dy, z) != 9) {
+            if (get_data(state, BLOCKS, x, dy, z) != 8 && get_data(state, BLOCKS, x, dy, z) != 9) {
```
Comment (CounterPillow) [inline overviewer_core/src/primitives/cave.c]: Same here.

File: overviewer_core/src/primitives/cave.c (hunk 1 above)
Comment (dgrimreaper) [reply overviewer_core/src/primitives/cave.c]: No idea, not a C programmer ;).  Want them in temporary variables?

Comment (tswsl1989) [reply overviewer_core/src/primitives/cave.c]: > Is the compiler smart enough to optimise away those repeated function calls?
Best answer I can give to that without checking a compiled version is "maybe".
I'd probably err on the side of a (clearly named) temporary variable either way.

Comment (dgrimreaper) [reply overviewer_core/src/primitives/cave.c]: ok will do

**Comment by dgrimreaper** [comment]
Travis you suck.  Sigh pip fail.

Hopefully these temp var declarations are ok.

**Comment by dgrimreaper** [comment]
All done

**Comment by CounterPillow** [comment]
Merged, thanks for your contribution.

**Pull request merged**